### PR TITLE
ANDROID: optimization: faster getAll() 

### PR DESCRIPTION
**Pull request by mkdotcom**
Hello
On android, is there a reason to create an image file for each contact when calling getAll() ?
This PR just send back the imageUri, and then getAll() is very much faster.
It works fine for me on my app, but I don't know if it's ok or not for everyone.

### CODE REVIEW

**Comment by rt2zz** [comment]
I recall we switched to doing a file copy for some reason, but cannot remember said reason atm. @morenoh149 @npomfret any ideas?

Were we to take this change you will need to also remove the `getPhotoURIFromContactURI` method.

**Comment by morenoh149** [comment]
@mkdotcom is the `getAll` really faster this way? The problem previously was that too much memory was being consumed with your approach. This is a problem for large address books (thousands of contacts).

See previous discussions at https://github.com/rt2zz/react-native-contacts/pull/98 https://github.com/rt2zz/react-native-contacts/issues/59 and https://github.com/rt2zz/react-native-contacts/issues/72

@npomfret could you summarize the pros and cons of each approach

**Comment by mkdotcom** [comment]
i have 197 contacts and 42 of them have an image
it needs 0.5 seconds to load them with my PR and 8 seconds with "master"

**Comment by morenoh149** [comment]
It may also be a consequence of performing the get in a worker thread. See https://github.com/rt2zz/react-native-contacts/pull/109

**Comment by npomfret** [comment]
I suspect its the images.  I think we really need to change the API as discussed so that the _getting_ of the images is done separately to the getting of the contacts.  I'll try and find time to do it soon.

**Comment by npomfret** [comment]
... and I'm unsure as to why a copy was being made.  Maybe the uri returned isn't understood by RN?

**Comment by mkdotcom** [comment]
The url returned is fine (at least in the latest RN, with latest Android SDK and on my Android 4.4 device)
Maybe previously it wasn't, so that's why there was this image copy? (or on old Android versions?)
If we can't find out why there was the image copy, best idea could be to add a parameter to getAll() to choose if we want the copy or not

**Comment by npomfret** [comment]
I'd prefer to remove it.  See if anyone complains, we can always add it back.

**Comment by tszajna0** [comment]
I did something similar, also addressing iOS:
https://github.com/rt2zz/react-native-contacts/pull/118

@mkdotcom I think we overlapped a bit with this feature

**Comment by morenoh149** [comment]
Closing this in favor of #118. I think we could also have a more focused conversation over there over what the proper api should be for dealing with contacts and their photos. Along with related performance issues.

thanks @mkdotcom
